### ISSUE TITLE
[feat]: Skeleton component with auto-sizing

### ISSUE THREAD

**Issue by juliusmarminge** [shadcn-ui/ui#4608]
### Feature description

Radix Themes got this super nice skeleton component which has auto-sizing based on the children and a controllable state:

https://www.radix-ui.com/themes/docs/components/skeleton

Feels overkill installing Radix Themes just for this one component though

Usage:

```tsx
<Skeleton>
  {Number(500_000).toLocaleString()}
</Skeleton>
```

No need to manually specify the size, just put some placeholder data and the skeleton adjusts accordingly

With controllable state:

```tsx
<Skeleton loading={false}>
  {Number(500_000).toLocaleString()}
</Skeleton>
```

would display the actual number

### Affected component/components

Skeleton

### Additional Context

Additional details here...

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues and PRs

**Comment by viruop** [comment]
Hi @juliusmarminge I've made some changes. Could you please take a look and merge them? Thanks!

**Comment by juliusmarminge** [comment]
It seems to take up too much space:

https://github.com/user-attachments/assets/96b5db4c-b204-4bc6-84cb-9141beb56d41

I am really bad at all the implicit padding elements have but guessing there just has to be a class swap somehwere

**Comment by juliusmarminge** [comment]
Ah my bad, just gotta make sure the Skeleton wraps the text node and not any wrapping span or similar:


https://github.com/user-attachments/assets/da0e7154-45b0-400f-ba56-779dc8cff957

```diff
-    <Skeleton loading={!!props.loadingTotal}>
     <span className="text-lg font-bold leading-none sm:text-3xl">
+       <Skeleton loading={!!props.loadingTotal}>
          {chartConfig[props.chart].formatValue(props.total)}
+       </Skeleton>
      </span>
-  </Skeleton>
```

**Comment by viruop** [comment]
> Ah my bad, just gotta make sure the Skeleton wraps the text node and not any wrapping span or similar:
> 
>  CleanShot.2024-08-22.at.17.19.16.mp4 
> ```diff
> -    <Skeleton loading={!!props.loadingTotal}>
>      <span className="text-lg font-bold leading-none sm:text-3xl">
> +       <Skeleton loading={!!props.loadingTotal}>
>           {chartConfig[props.chart].formatValue(props.total)}
> +       </Skeleton>
>       </span>
> -  </Skeleton>
> ```

Yes, the Skeleton should only wrap the text node to avoid unnecessary padding from other elements.

**Comment by viruop** [comment]
Hi @juliusmarminge, are there any further changes required from my end? If not, can you please merge this PR (https://github.com/shadcn-ui/ui/pull/4620)

**Comment by juliusmarminge** [comment]
I am not a maintainer of this repo.

**Comment by viruop** [comment]
Glad it worked for you! I'm happy to help

**Comment by ayyjohn** [comment]
same as @juliusmarminge just popping in to say I tried it out and it works great

**Comment by viruop** [comment]
@Rick-Phoenix Thanks for sharing this, great implementation!

**Comment by shadcn** [comment]
This issue has been automatically marked as stale due to two years of inactivity. It will be closed in 7 days unless there’s further input. If you believe this issue is still relevant, please leave a comment or provide updated details. Thank you.

**Comment by shadcn** [comment]
Not stale. Something went wrong with the stale bot.

**Comment by shadcn** [comment]
This issue has been automatically marked as stale due to one year of inactivity. It will be closed in 7 days unless there’s further input. If you believe this issue is still relevant, please leave a comment or provide updated details. Thank you. (This is an automated message)

**Comment by viruop** [comment]
Hi @shadcn, this feature request is still relevant.

I had implemented this earlier in PR (https://github.com/shadcn-ui/ui/pull/4620), but it appears the PR was automatically closed due to inactivity. The implementation should still be valid.

Could this PR be reopened for review, or would you prefer that I create a new PR with the same changes?
